clients/vault/src/oracle/collector/proof_builder.rs

```diff
@@ -128,8 +128,11 @@ impl ScpMessageCollector {
 		sender: &StellarMessageSender,
 	) -> Option<UnlimitedVarArray<ScpEnvelope>> {
 		if let Some(envelopes) = self.envelopes_map().get(&slot) {
-			// lacking envelopes
-			if envelopes.len() < get_min_externalized_messages(self.is_public()) {
+			// Lacking envelopes.
+			// If the data was provided from the archive, no need to check for the minimum
```

Comment:
```suggestion
			// If the data was provided from the archive, no need to check for the minimum
			// Otherwise, we are still lacking envelopes.
```

Reply:
done